8dev_restapi.js

```diff
@@ -37,6 +37,30 @@ class Endpoint extends EventEmitter {
     });
   }
 
+  read(path, callback) {
+    return new Promise((fulfill, reject) => {
+      this.service.get('/endpoints/' + this.id + path, (data, resp) => {
+        if (resp.statusCode === 202) {
+          this.service.on('async-response', (asyncResponse) => {
+            if (data['async-response-id'] === asyncResponse['id']) {
+              callback(asyncResponse.payload);
```

Comment:
Callback should be the same as with observe. It should either be callback(code, payload) or callback(response). In latter case the observe and write callbacks should be changed too.